### DISCUSSION TITLE
Add prometheusModule for cephcluster crds?

### DISCUSSION THREAD

**Discussion by thenamehasbeentake** [rook/rook#11942] [Ideas]
Mgr prometheusModule create by default, and can't update or delete by crds. Should we add `PrometheusSpec` into `ClusterSpec` like `DashboardSpec`?
https://github.com/rook/rook/blob/6273bc1aec7087465829195a89378abba7d5ec8a/pkg/apis/ceph.rook.io/v1/types.go#L182

**Comment by travisn** [comment]
Are you looking for the MonitoringSpec type? If you want prometheus enabled, then you'll need to set `monitoring.enabled: true`.

**Reply by thenamehasbeentake** [reply]
I mean `ceph mgr module`. you can run cmd in toolbox like this:
~~~
ceph mgr services
{
    "dashboard": "http://10.9.8.102:8080/",
    "prometheus": "http://10.9.8.102:9284/"
}
~~~
`prometheus`  module is always enabled and it can't close by crds
  

enablePrometheusModule code is here:
https://github.com/rook/rook/blob/6273bc1aec7087465829195a89378abba7d5ec8a/pkg/operator/ceph/cluster/mgr/mgr.go#L321

**Reply by travisn** [reply]
I see, the prometheus module is always enabled. If you want to disable the prometheus module, the operator should look at the setting for `monitoring.enabled: false` and disable the module in that case. Want to open a new issue for that?

**Reply by thenamehasbeentake** [reply]
https://github.com/rook/rook/issues/11962
